stdlib/LinearAlgebra/src/triangular.jl

```diff
@@ -979,9 +979,20 @@ _trimul!(C::AbstractMatrix, A::UpperOrLowerTriangular, B::AbstractTriangular) =
 _trimul!(C::AbstractMatrix, A::AbstractTriangular, B::UpperOrLowerTriangular) =
     generic_mattrimul!(C, uplo_char(B), isunit_char(B), wrapperop(parent(B)), A, _unwrap_at(parent(B)))
 
-lmul!(A::AbstractTriangular, B::AbstractVecOrMat) = @inline _trimul!(B, A, B)
-rmul!(A::AbstractMatrix, B::AbstractTriangular)   = @inline _trimul!(A, A, B)
-
+function lmul!(A::AbstractTriangular, B::AbstractVecOrMat)
+    if istriu(A)
```

Comment:
I wonder if we should have one-arg `istriu(::UpperTriangular)` (and `istril` correspondingly) to compile away the checks and the "rewrap" for `UpperTriangular`?

Reply:
The current implementation should compile away because of constant propagation

Reply:
I couldn't see it, when I checked.

Reply:
IIUC this is what you're referring to?
```julia
julia> @code_typed istriu(UpperTriangular(zeros(3,3)))
CodeInfo(
1 ─     return true
) => Bool

julia> @code_typed istril(LowerTriangular(zeros(3,3)))
CodeInfo(
1 ─     return true
) => Bool
```

Perhaps we may define the single-argument methods as well, so that this works even if constant-propagation fails for some reason.

Reply:
Ah, sorry for the noise, I checked quickly on v1.10, but you may have enforced constant propagation meanwhile.

Reply:
Yes, the aggressive constprop has been added to nightly, but it's not present on v1.11 or lower.